src/runtime_src/tools/xclbinutil/Section.cxx

```diff
@@ -181,6 +181,21 @@ Section::supportsSectionIndex(enum axlf_section_kind &eKind)
   return iter->get()->supportsIndexing;
 }
 
+bool 
+Section::supportsSubSectionName(enum axlf_section_kind eKind, const std::string & sSubSectionName)
+{
+  auto& sections = getSectionTypes();
+  auto sectionIter = std::find_if(sections.begin(), sections.end(), [&](const auto &entry) { return entry->eKind == eKind; });
+
+  if (sectionIter == sections.end())
+    return false;
+
+  auto subSections = sectionIter->get()->subSections;
+
+  return std::any_of(subSections.begin(), subSections.end(), [&](const auto &entry) { return boost::iequals(entry, sSubSectionName); });
+}
```

Comment:
To bad we cannot use C++20. All this function would be 1 line. :-/

Reply:
Agree 100%.  Can't wait till we offically support this version of C++.